python/ray/dataframe/indexing.py

```diff
@@ -1,104 +1,431 @@
+
+        # iloc[1,:] returns a row series: row 1
+        if is_scalar(row_loc) and is_2d(col_loc):
+            return self._get_row_series(row_loc, col_loc)
+
+        # iloc[:,1] returns a columns series: col 1
+        if is_2d(row_loc) and is_scalar(col_loc):
+            return self._get_col_series(row_loc, col_loc)
+
+        if is_2d(row_loc) and is_2d(col_loc):
+            return self._get_dataframe_view(row_loc, col_loc)
+
+    def _get_scaler(self, row_loc, col_loc):
         pass
 
-    def _map_partition(self, lookup_dict, col_lst, indexer='loc'):
-        """Apply retrieval function to a lookup_dict
-        in the form of {partition_id: [idx]}.
+    def _get_col_series(self, row_loc, col_loc):
+        pass
 
-        Returns:
-            retrieved_rows_remote: a list of object ids for pd_df
-        """
-        assert indexer in ['loc', 'iloc'], "indexer must be loc or iloc"
+    def _get_row_series(self, row_loc, col_loc):
+        pass
+
+    def _get_dataframe_view(self, row_loc, col_loc):
+        pass
 
-        if indexer == 'loc':
 
-            def retrieve_func(df, idx_lst, col_label):
-                return df.loc[idx_lst, col_label]
-        elif indexer == 'iloc':
+def is_enlargement(locator, corrd_df):
+    # Enlargement happens when you trying to locate using labels isn't in the
+    # original index. In other words, enlargement == adding NaNs !
+    if is_list_like(locator) and not is_slice(
+            locator) and len(locator) > 0 and not is_bool(locator[0]):
+        n_diff_elems = len(pd.Index(locator).difference(corrd_df.index))
+        is_enlargement_boolean = n_diff_elems > 0
+        if is_enlargement_boolean:
+            warn(
+                FutureWarning("""
+Passing list-likes to .loc or [] with any missing label will raise
+KeyError in the future, you can use .reindex() as an alternative.
 
-            def retrieve_func(df, idx_lst, col_idx):
-                return df.iloc[idx_lst, col_idx]
+See the documentation here:
+http://pandas.pydata.org/pandas-docs/stable/indexing.html#deprecate-loc-reindex-listlike
+            """))
 
-        retrieved_rows_remote = [
-            _deploy_func.remote(retrieve_func,
-                                self.df._row_partitions[partition],
-                                idx_to_lookup, col_lst)
-            for partition, idx_to_lookup in lookup_dict.items()
-        ]
-        return retrieved_rows_remote
+        return is_enlargement_boolean
+    return False
 
 
 class _Loc_Indexer(_Location_Indexer_Base):
     """A indexer for ray_df.loc[] functionality"""
 
-    def locate_2d(self, row_label, col_label):
-        index_loc = self.df._row_index.loc[row_label]
-        lookup_dict = self._get_lookup_dict(index_loc)
-        retrieved_rows_remote = self._map_partition(
-            lookup_dict, col_label, indexer='loc')
-        joined_df = pd.concat(ray.get(retrieved_rows_remote))
+    def _get_scaler(self, row_loc, col_loc):
```

Comment:
Would you be able to add some comments about what these methods do (this and methods below)? Just so that we can quickly look at the code and maintain it better.